datafusion/datasource-parquet/src/opener.rs

```diff
@@ -111,19 +120,61 @@ impl FileOpener for ParquetOpener {
             .create(projected_schema, Arc::clone(&self.table_schema));
         let predicate = self.predicate.clone();
         let table_schema = Arc::clone(&self.table_schema);
+        let partition_fields = self.partition_fields.clone();
         let reorder_predicates = self.reorder_filters;
         let pushdown_filters = self.pushdown_filters;
         let coerce_int96 = self.coerce_int96;
         let enable_bloom_filter = self.enable_bloom_filter;
         let enable_row_group_stats_pruning = self.enable_row_group_stats_pruning;
         let limit = self.limit;
 
-        let predicate_creation_errors = MetricBuilder::new(&self.metrics)
-            .global_counter("num_predicate_creation_errors");
-
         let enable_page_index = self.enable_page_index;
 
         Ok(Box::pin(async move {
+            // Prune this file using the file level statistics.
```

Comment:
I worry that in the case when there aren't any dynamic predicates, trying to prune again on file opening is simply going to be pure overhead / wasteful.

Therefore, I think it would be good if we could somehow control / disable trying to apply this extra filtering when it is known it would not help

For example, maybe we can have a field on `ParquetOpener` with something like `prune_on_open` which can be set to true if there are dynamic predicates present.

This would also likely ensure the tests can pass again

Reply:
> which can be set to true if there are dynamic predicates present

the issue is: how do we know the filters are dynamic? we've hidden dynamic filters behind `PhysicalExpr` so that the system can treat them as normal filters.  we could do _any_ filter pushdown but that doesn't seem like much of an improvement.

I also think this pruning should be quite cheap / the record batches being filtered are just a couple rows